packages/common/react-components/src/components/Select/Select.tsx

```diff
@@ -0,0 +1,96 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import * as SelectPrimitive from '@radix-ui/react-select';
+import { CaretDown, CaretUp, Check } from 'phosphor-react';
+import React from 'react';
+
+import { mx } from '../../util';
+import { Button } from '../Button';
+
+export type OptionObject = { id?: string; title: string; disabled?: boolean };
+export type Option = string | OptionObject;
+
+export type SelectProps = {
+  defaultValue?: string;
+  label?: string;
+  options?: Option[];
+  onChange?: (value: string, item?: Option) => any;
+};
+
+const value = (item: Option) => (typeof item === 'string' ? item : item.id ?? item.title);
+const text = (item: Option) => (typeof item === 'string' ? item : item.title ?? item.id);
+const disabled = (item: Option) => (typeof item === 'string' ? false : item.disabled);
+
+const eq = (a: Option, b: Option) =>
+  a &&
+  b &&
+  (a === b || (typeof a !== 'string' && typeof b !== 'string')
+    ? (a as OptionObject).id === (b as OptionObject).id
+    : (a as OptionObject).id === b || (b as OptionObject).id === a);
+
+const defaultOption = (items: Option[], defaultValue?: Option) => {
+  if (!items) {
+    return undefined;
+  }
+  if (!defaultValue) {
+    return items?.[0];
+  }
+  return items.find((s) => eq(s, defaultValue)) ?? items[0];
+};
+
+export const Select = (props: SelectProps) => {
+  const { label, options, onChange } = { options: [], ...props };
+  return (
+    <SelectPrimitive.Root
+      defaultValue={value(defaultOption(options, props.defaultValue) ?? '')}
+      onValueChange={(v) =>
+        onChange?.(
+          v,
+          options.find((s) => eq(s, v))
+        )
+      }
+    >
+      <SelectPrimitive.Trigger asChild aria-label={label}>
+        <Button>
+          <SelectPrimitive.Value />
+          <SelectPrimitive.Icon className='ml-2'>
+            <CaretDown />
+          </SelectPrimitive.Icon>
+        </Button>
+      </SelectPrimitive.Trigger>
+      <SelectPrimitive.Content>
+        <SelectPrimitive.ScrollUpButton className='flex items-center justify-center text-gray-700 dark:text-gray-300'>
+          <div className=''>
+            <CaretUp />
+          </div>
+        </SelectPrimitive.ScrollUpButton>
+        <SelectPrimitive.Viewport className='bg-white dark:bg-gray-800 p-2 rounded-lg shadow-lg'>
+          <SelectPrimitive.Group>
```

Comment:
I was imagining we’d export this separately, I think the developer should assemble groups and options together rather than using props.

Reply:
I say this mainly because the mental model for this package is in-flux, and for ergonomic reasons we should settle on at least a rule of thumb: if it’s a collection, the collection items are components you render as children.

Reply:
for now i'm on the side of doing that when a new usage pops up requiring groups (and we can also make a minor effort to avoid needing that in the product for a while ;) )

Reply:
we agreed to use children and it looks nicer, but i witheld adding some slots people might want in the Select.Items for later when someone needs it